src/workerd/api/basics.h

```diff
@@ -609,15 +620,53 @@ class AbortSignal final: public EventTarget {
     tracker.trackField("reason", reason);
   }
 
+  void serialize(jsg::Lock& js, jsg::Serializer& serializer);
+
+  // To test what happens if a capability is dropped before invoking release on the cloned abort
+  // signal, this method will tell every rpcClient to skip this step before destruction.
+  void skipReleaseForTest();
+
+  static jsg::Ref<AbortSignal> deserialize(
+      jsg::Lock& js, rpc::SerializationTag tag, jsg::Deserializer& deserializer);
+
+  JSG_SERIALIZABLE(rpc::SerializationTag::ABORT_SIGNAL);
+
  private:
   IoOwn<RefcountedCanceler> canceler;
   Flag flag;
   kj::Maybe<jsg::JsRef<jsg::JsValue>> reason;
   kj::Maybe<jsg::JsRef<jsg::JsValue>> onAbortHandler;
 
+  // Dispatch an abort event to all registered listeners (including RPC and local event handlers)
+  void dispatchEvent(jsg::Lock& js);
+
   void visitForGc(jsg::GcVisitor& visitor);
 
   friend class AbortController;
+
+  /** RPC client functionality. Used if this signal was serialized. **/
+
+  // A collection of rpcClients, which will be notified if this signal is triggered and when this
+  // signal is destroyed.
+  kj::Vector<IoOwn<AbortTriggerRpcClient>> rpcClients;
+
+  // Trigger an abort on all associated clients
+  kj::Promise<void> sendToRpc(kj::Array<kj::byte>&& reason);
+
+  /** RPC server functionality. Used if this signal was deserialized. **/
+
+  // A promise that is fulfilled if an abort() message is received over RPC.
+  kj::Maybe<kj::Promise<void>> maybeRpcAbortPromise;
+
+  // A refcounted object used to receive a serialized abort reason
+  kj::Own<PendingReason> pendingReason;
```

Comment:
You will need this to be `IoOwn` since `PendingReason` may be modified on the RPC's I/O thread without holding the isolate lock.